scarb/src/compiler/plugin/proc_macro/host.rs

```diff
@@ -1172,3 +1173,66 @@ impl ProcMacroHost {
         &self.macros
     }
 }
+
+/// A global storage for dynamically-loaded procedural macros.
+/// Loads dynamic shared libraries and hides them beside [`ProcMacroHostPlugin`].
+/// Guarantees that every library is loaded exactly once,
+/// but does not prevent loading multiple versions of the same library.
+#[derive(Default)]
+pub struct ProcMacroRepository {
+    /// A mapping between the [`PackageId`] of the package which defines the plugin
+    /// and the [`ProcMacroHostPlugin`] holding the underlying shared library.
+    macros: HashMap<PackageId, Arc<ProcMacroHostPlugin>>,
```

Comment:
It's not extremely obvious for me, whether we should have 
```
proc macro expansion - many to 1 -> proc macro instance - 1 to 1 -> proc macro host plugin - many to 1 -> single package cairo plugin dependencies
```
or 
``` 
proc macro expansion - many to 1 -> proc macro instance - many to 1 -> proc macro host plugin - 1 to 1 -> single package cairo plugin dependencies
```
(or any other configuration)
Can you elaborate on the decision made here?